public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -40,6 +43,25 @@ export function QueryTemplatesList(props: QueryTemplatesListProps) {
         return;
       }
 
+      let userDataList;
+      try {
+        const userQtList = uniq(data.map((qt) => qt.user?.uid));
+        const usersParam = userQtList.map((user) => `key=${user}`).join('&');
+        userDataList = await getUserInfo(`?${usersParam}`);
```

Comment:
I've tested sending over hundred of user uids in one batch and didn't run into any issues. The response time was ~16ms